R/methods-Network.R

```diff
@@ -0,0 +1,231 @@
+# Methods for the Network class
+
+#' @include methods-Nodes.R
+# Fancy accessors
+setGeneric("getNodes", function(object) standardGeneric("getNodes"))
+setMethod("getNodes", "Network", function(object) object@nodes)
+setMethod("getNodeIds", "Network", function(object) getNodeIds(object@nodes))
+setGeneric("getLinks", function(object) standardGeneric("getLinks"))
+setMethod("getLinks", "Network", function(object) object@links)
+setGeneric("getLinkColorScheme", function(object) standardGeneric("getLinkColorScheme"))
+setMethod("getLinkColorScheme", "Network", function(object) object@linkColorScheme)
+# No setters! Once created, a network should only be updated via network methods
+
+
+## General network methods
+
+#' Get isolated nodes
+#' 
+#' Returns a list of nodes that have no links
+#' @param net A Network object
+#' @export
+setGeneric("getIsolatedNodes", function(net) standardGeneric("getIsolatedNodes"))
+
+#' @export
+setMethod("getIsolatedNodes", "Network", function(net) {
+  nodes <- getNodes(net)
+  links <- getLinks(net)
+
+  nodesWithLinks <- NodeList(union(getSourceNodes(links), getTargetNodes(links)))
+  isolatedNodeIds <- setdiff(getNodeIds(nodes), getNodeIds(nodesWithLinks))
+  isolatedNodes <- NodeList(nodes[which(getNodeIds(nodes) %in% isolatedNodeIds)])
+
+  return(isolatedNodes)
+})
+
+
+#' Remove isolated nodes
+#' 
+#' Removes nodes that have no links
+#' @param net A Network object
+#' @param verbose If TRUE, will print messages
+#' @export
+setGeneric("pruneIsolatedNodes", function(net, verbose = c(TRUE, FALSE)) standardGeneric("pruneIsolatedNodes"))
+
+#' @export
+setMethod("pruneIsolatedNodes", "Network", function(net, verbose = c(TRUE, FALSE)) {
+  verbose <- veupathUtils::matchArg(verbose)
+  nodes <- getNodes(net)
+  isolatedNodeIds <- getNodeIds(getIsolatedNodes(net))
+
+  if (length(isolatedNodeIds) > 0) {
+    net@nodes <- nodes[which(!getNodeIds(nodes) %in% isolatedNodeIds)]
+    validObject(net)
+    veupathUtils::logWithTime(paste('Found and removed', length(isolatedNodeIds), 'isolated nodes.'), verbose)
+  } else {
+    veupathUtils::logWithTime('No isolated nodes found.', verbose)
+  }
+
+  validObject(net)
+  return(net)
+})
+
+getLinkUniqueString <- function(link) {
+  paste0(id(source(link)), id(target(link)))
```

Comment:
OOoh good idea. We should add a funky delimiter between the two so that the following two links don't get counted as duplicates:
1. Source: "A", Target: "BC"
2. Source: "AB", "Target: "C"